src/dom-manipulation.js

```diff
@@ -7,20 +8,32 @@
  * only spot in the codebase with impure functions.
  */
 import { findDOMNode } from 'react-dom';
-
-
-export function applyStylesToDOMNode({ domNode, styles }) {
+import type { Component } from 'react';
+import type {
+  Styles,
+  ClientRect,
+  GetPosition,
+  NodeData,
+  VerticalAlignment,
+  ConvertedProps,
+} from './typings';
+
+export function applyStylesToDOMNode({ domNode, styles }: {
+  domNode: HTMLElement,
+  styles: Styles,
+}) {
   // Can't just do an object merge because domNode.styles is no regular object.
   // Need to do it this way for the engine to fire its `set` listeners.
   Object.keys(styles).forEach((key) => {
-    // eslint-disable-next-line no-param-reassign
-    domNode.style[key] = styles[key];
+    const value = styles[key];
+    const stringValue = typeof value === 'number' ? `${value}px` : value;
+    domNode.style.setProperty(key, stringValue);
```

Comment:
Hah, you ran into this guy: https://github.com/facebook/flow/issues/3733

(I started working on this issue myself, but I didn't get much farther than this).

I like your solution, but I see a small issue with it: the style being set might not have a pixel value. For example, `opacity` is something often used in enter/leave animations.

Could we not just do:

```js
const value = styles[key];
domNode.style.setProperty(key, String(value));
```

Off the top of my head, I can't think of any reason that this wouldn't work, but it should definitely be tested with the included stories (and maybe even adding one that defines a custom enterAnimation that animates `height` or something equally heinous)